### DISCUSSION TITLE
Why is the NumberInput value a string?

### DISCUSSION THREAD

**Discussion by toniengelhardt** [chakra-ui/ark#2527] [Q&A] [answered]
Is there any good reason that the `NumberInput.Root` value is a string? Would it not make more sense that this abstraction over the native input field takes and emits a `number`? For context I'm using `@ark-ui/vue`. Thanks!

**Comment by cschroeter** [comment]
Good question. I think we could support both just as the DOM native input. Any objection @segunadebayo ?

**Reply by toniengelhardt** [reply]
I reckon input `type=text` was chosen to support formatting for large numbers, but the root element could convert the value to number for us. Otherwise, I guess 95% of users will have to implement the logic by themselves (and it's a bit fizzly). But for me `type=number` would also work perfectly fine.

**Reply by segunadebayo** [reply, marked as answer]
We can't support this for two reasons:

- There's an option to format the number input's value based on the Intl.FormatOptions
- User might run into NaN scenarios when typing decimal dot, say "12."

Hope this helps

**Reply by toniengelhardt** [reply]
Thanks 🙏🏽
